File: auto-instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/internal/RequestPropertySetter.java
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.library.httpurlconnection.internal;

import io.opentelemetry.context.propagation.TextMapSetter;
import java.net.URLConnection;

public enum RequestPropertySetter implements TextMapSetter<URLConnection> {
    INSTANCE;

    @Override
    public void set(URLConnection carrier, String key, String value) {
        carrier.setRequestProperty(key, value);
    }
}

File: instrumentation/httpurlconnection/library/src/main/java/io/opentelemetry/instrumentation/library/httpurlconnection/internal/RequestPropertySetter.kt
/*
 * Copyright The OpenTelemetry Authors
 * SPDX-License-Identifier: Apache-2.0
 */

package io.opentelemetry.instrumentation.library.httpurlconnection.internal

import io.opentelemetry.context.propagation.TextMapSetter
import java.net.URLConnection

internal object RequestPropertySetter : TextMapSetter<URLConnection?> {
    override fun set(
        carrier: URLConnection?,
        key: String,
        value: String,
    ) {
        carrier?.setRequestProperty(key, value)
    }
}
